pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -562,6 +562,12 @@ func (s *Reconciler) createNetworkInterface(ctx context.Context, nicName string)
 
 		skuI, err := s.resourcesSkus.Get(ctx, skuSpec)
 		if err != nil {
+			metrics.RegisterFailedInstanceCreate(&metrics.MachineLabels{
+				Name:      s.scope.Machine.Name,
+				Namespace: s.scope.Machine.Namespace,
+				Reason:    err.Error(),
+			})
+
```

Comment:
Do we normally do this for errors or just invalid machine config? Wondering if this should only be in the first condition of the next if statement, WDYT?

Reply:
We increment the metric when we call Azure API with a request needed to create machine that return error e.g. creating machine, IP, interface. We do this for all types of errors. Not just invalid config.